File: Programmers/pro_12913_땅따먹기.py
def solution(land):
    for r in range(1,len(land)):
        for c in range(4):
            land[r][c]=land[r][c]+max(land[r-1][(c+1)%4],land[r-1][(c+2)%4],land[r-1][(c+3)%4])
    return max(land[-1])

File: Programmers/pro_12913_땅따먹기.js
function solution(land) {
    const len = land.length
    for (let i = 1; i < len; i++) {
        for (let j = 0; j < 4; j++) {
            land[i][j] += Math.max(land[i - 1][(j + 1) % 4], land[i - 1][(j + 2) % 4], land[i - 1][(j + 3) % 4])
        }
    }
    return Math.max(...land[len - 1]);
}